peer/upstream/setup.go

```diff
@@ -234,19 +237,21 @@ func (u *Upstream) requestConnect() error {
 	}
 }
 
-// must have lock held before calling this
```

Comment:
The comment is pertinent, I think you should keep the comment and add XXX: at the beginning of it as a warning for other developers to know a lock is needed.

Reply:
Since i changed the lock held in this method by locking reading `u.client`, so i think this comment is not true, it maybe makes developer easier to get double locking when calling this method.

Reply:
Yes makes sense, I didn't see by this perspective.